src/librustc/ty/structural_impls.rs

```diff
@@ -95,8 +96,11 @@ impl fmt::Debug for ty::BoundRegion {
         match *self {
             ty::BrAnon(n) => write!(f, "BrAnon({:?})", n),
             ty::BrNamed(did, name) => {
-                write!(f, "BrNamed({:?}:{:?}, {})",
-                        did.krate, did.index, name)
+                if did.index == CRATE_DEF_INDEX {
```

Comment:
I think `did.is_local()` is the canonical (and to me easier to read) way to express this

Reply:
That's `did.krate == LOCAL_CRATE` though, this is about the def inside the crate (whichever crate it may be).

Reply:
I knew something was off